components/cdc/src/observer.rs

```diff
@@ -119,15 +121,16 @@ impl<E: KvEngine> CmdObserver<E> for CdcObserver {
         let mut region = Region::default();
         region.mut_peers().push(Peer::default());
         // Create a snapshot here for preventing the old value was GC-ed.
-        // TODO: only need it after enabling old value, may add a flag to indicate
-        // whether to get it.
         let snapshot = RegionSnapshot::from_snapshot(Arc::new(engine.snapshot()), Arc::new(region));
         let get_old_value = move |key,
                                   query_ts,
                                   old_value_cache: &mut OldValueCache,
                                   statistics: &mut Statistics| {
             old_value::get_old_value(&snapshot, key, query_ts, old_value_cache, statistics)
         };
+
+        let size = cmd_batches.iter().map(|b| b.size()).sum();
+        self.memory_quota.alloc_force(size);
```

Comment:
Please add a test that ensures that too many pending `CmdBatch` causes `TxnExtra` to be dropped due to memory quota.